specification/riskiq/data-plane/Microsoft.Easm/preview/2023-03-01-preview/easm.json

```diff
@@ -0,0 +1,5656 @@
+          "Discovery Groups"
+        ],
+        "operationId": "DiscoveryGroups_List",
+        "summary": "Retrieve a list of discovery group for the provided search parameters.",
+        "x-ms-examples": {
+          "DiscoveryGroups": {
+            "$ref": "./examples/DiscoveryGroups_List.json"
+          }
+        },
+        "x-ms-pageable": {
+          "itemName": "content",
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/workspaces/{workspaceName}/discoGroups/{groupName}": {
+      "get": {
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DiscoGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/DiscoGroupResponse"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "headers": {
+              "x-ms-error-code": {
+                "type": "string"
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "tags": [
+          "Discovery Groups"
+        ],
+        "operationId": "DiscoveryGroups_Get",
+        "summary": "Retrieve a discovery group with a given groupName.",
+        "x-ms-examples": {
+          "DiscoveryGroups": {
+            "$ref": "./examples/DiscoveryGroups_Get.json"
+          }
+        }
+      },
+      "put": {
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DiscoGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DiscoGroupRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/DiscoGroupResponse"
+            }
+          },
```

Comment:
For a PUT, wondering whether same roundtrip model can be shared by both request and response (if property only in response, mark as readonly).

Reply:
Can you please elaborate on or give me a documentation for the roundtrip model?

Reply:
https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#rest-response-body-is-resource-schema

example (just click the first one I found) https://github.com/Azure/azure-rest-api-specs/blob/main/specification/purview/data-plane/Azure.Analytics.Purview.Account/preview/2019-11-01-preview/account.json#L225-L243

Reply:
I don't think we'd use roundtrip model here since there are a few properties only available in response, but we'll definitely mark them as readonly

Reply:
"readonly" do mean customer cannot set the value, which is the standard solution to the case of "a few properties only available in response".

Every single ARM resource is such, that `id` property is readonly, and cannot set by user.

Reply:
You should only mark properties as readOnly if they are in a schema that could be sent in a request body. If the schema is only used for responses, there is no value in marking properties readOnly.